core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala

```diff
@@ -160,7 +160,7 @@ protected[controller] class RestAPIVersion(apipath: String, apiversion: String)(
             sendCorsHeaders {
                 (pathEndOrSingleSlash & get) {
                     complete(OK, info)
-                } ~ authenticate(basicauth) {
+                } ~ (authenticate(basicauth) | authenticate(certificateAuth)) {
```

Comment:
The routing is an "or" route. I can supply a bad key or no key and specify any namespace and get access to it. How do you prevent that?